core/xml.js

```diff
@@ -462,9 +407,178 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
     workspace.setResizesEnabled(true);
   }
   Blockly.Events.fire(new Blockly.Events.FinishedLoading(workspace));
-  return newBlockIds;
+  return state.newBlockIds;
 };
 
+/**
+ * Defines the behavior for how the parser should handle errors. Projects using
+ * Blockly can override this method to provide project-specific error handling
+ * or to present more meaningful information to users. The default behavior is
+ * simply to throw the error.
+ * @param {!string} message The error message describing what went wrong.
+ * @public
+ */
+Blockly.Xml.onError = function(message) {
+  throw Error(message);
+};
+
+/**
+ * Defines the behavior for how the parser should handle warnings. Projects
+ * using Blockly can override this method to provide project-specific warning
+ * handling or to present more meaningful information to users. The default
+ * behavior is to log the message to the browser console.
+ * @param {!string} message The warning message describing what went wrong.
+ * @public
+ */
+Blockly.Xml.onWarning = function(message) {
+  console.warn(message);
+};
+
+/**
+ * A mapping of XML tags to parser functions.
+ * @type {Object.<string,
+ *     function(Blockly.Workspace, Element, Blockly.Xml.ParserState)>}
+ */
+Blockly.Xml.tagMap_ = {};
+
+/**
+ * A list of operations to serialize a workspace to XML.
+ * @type {Array.<function(Element, Blockly.Workspace, boolean)>}
+ * @private
+ */
+Blockly.Xml.serializers_ = [];
+
+/**
+ * The function to handle unrecognized tags when parsing an XML document.
+ * @param {!Blockly.Workspace} workspace the Blockly workspace
```

Comment:
The descriptions on param lines should start with a capital letter and end with punctuation (unless the start or end are code).  Thus:
@param {!Blockly.Workspace} workspace The Blockly workspace.

There are several more instances of this is this PR.